rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -92,22 +96,98 @@ def _process_deposit_or_withdrawal(
+                    notes=next_event.notes if next_event.notes else '',
+                    location=next_event.location,
+                    timestamp=next_event.get_timestamp_in_sec(),
+                    asset=next_event.asset,
+                    amount=next_event.balance.amount,
+                    taxable=False,  # Deposits and withdrawals are not taxable
+                    count_entire_amount_spend=False,
+                    count_cost_basis_pnl=False,
+                    extra_data={'tx_hash': next_event.serialized_event_identifier},
+                )
+            else:  # can only be acquisition
+                amount_deposited[next_event.asset] -= next_event.balance.amount
+                if amount_deposited[next_event.asset] < ZERO:
+                    profit = -1 * amount_deposited[next_event.asset]
+                    crypto_asset = next_event.asset.resolve_to_crypto_asset()
+                    pot.add_acquisition(
+                        event_type=AccountingEventType.TRANSACTION_EVENT,
+                        notes=f'Gained {profit} from removing {next_event.balance.amount} {crypto_asset.symbol} from {next_event.counterparty} pool',  # noqa: E501
+                        location=next_event.location,
+                        timestamp=next_event.get_timestamp_in_sec(),
+                        asset=event.asset,
+                        amount=profit,
+                        taxable=True,
+                        extra_data={'tx_hash': next_event.serialized_event_identifier},
+                    )
+                    amount_deposited[next_event.asset] = ZERO
+
+
+class TaxableLiquidityPoolAccountantInterface(ModuleAccountantInterface):
+    """Interface for protocols that support addition & removal of liquidity from pools.
+
+    It checks if
+    Such protocols are:
+    - Curve
+    - Yearn
+    - Uniswap v2 & v3
+    - Sushiswap v2
+    - Aave v1 & v2
+    - Convex
+    - Balancer v1
+    """
+    amount_deposited: dict[Asset, FVal] = defaultdict(FVal)
+
+    def reset(self) -> None:
+        self.amount_deposited = defaultdict(FVal)
+
+    def _process_liquidity_addition(
+            self,
+            pot: 'AccountingPot',  # pylint: disable=unused-argument
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],  # pylint: disable=unused-argument
+    ) -> None:
+        """Protocols can override this method."""
+        self.amount_deposited[event.asset] += event.balance.amount
+
+    def _process_liquidity_removal(
+            self,
+            pot: 'AccountingPot',  # pylint: disable=unused-argument
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],
+    ) -> None:
+        """Protocols can override this method."""
+        for other_event in other_events:
+            if (
+                other_event.event_type == HistoryEventType.WITHDRAWAL and
+                other_event.event_subtype == HistoryEventSubType.REMOVE_ASSET and
```

Comment:
I agree that we should have those standardized but you should document this as requirement in the docstring.